env/work/srw_python/srwlib.py

```diff
@@ -3844,8 +3847,45 @@ def srwl_uti_save_stat_wfr_emit_prop_multi_e(  # #MR20160908
         'status': status,
     }
     status_json_file = '{}.json'.format(filename)
-    with open(status_json_file, 'w') as f:
-        json.dump(status, f, indent=4, separators=(',', ': '), sort_keys=True)
+    #MR05122016: A temp file is created on the same filesystem as the status file to ensure the atomic rename operation:
+    # See the following discussions:
+    # - https://github.com/radiasoft/sirepo/issues/555
+    # - https://github.com/radiasoft/SRW-light/commit/987547f4da3079626bef92aad2f9ca3bade84ada
+    # - http://stackoverflow.com/a/3716361/4143531
+    tmp_file = tempfile.NamedTemporaryFile(delete=False, mode='w', dir=os.path.dirname(status_json_file))
+    json.dump(status, tmp_file, indent=4, separators=(',', ': '), sort_keys=True)
+    tmp_file.close()
+    shutil.move(tmp_file.name, status_json_file)
+
+def srwl_uti_save_stat_wfr_emit_prop_multi_e_init(rank, num_of_proc, num_part_per_proc, num_sent_per_proc, num_part_avg_proc): #MR20012017
+    """Initialize parameters for the SRW status files and generate the files.
+
+    :param rank: rank of the process.
+    :param num_of_proc: total number of processes.
+    :param num_part_per_proc: number of electrons treated by each worker process.
+    :param num_sent_per_proc: number of sending acts made by each worker process.
+    :param num_part_avg_proc: number of macro-electrons to be used in calculation by each worker.
+    """
+    log_dir = os.path.abspath('__srwl_logs__')
```

Comment:
'__srwl_logs__' could be made a default value of some input variable. Maybe later.